### PR TITLE
Add the NASA_DAAC_download function

### PR DESCRIPTION
**Pull request by DongchenZ**
## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/data.remote/R/NASA_DAAC_download.R [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom foreach %dopar%
+NASA_DAAC_download <- function(ul_lat,
+                               ul_lon,
+                               lr_lat,
+                               lr_lon,
+                               ncore = parallel::detectCores() - 1,
```
Comment (infotroph) [inline modules/data.remote/R/NASA_DAAC_download.R:29]: This should default to 1 -- multicore needs the user's explicit request.

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ see if you need to change any of these:
 - Added new SDA workflow for the 342 North America anchor sites.
 - Added new feature of preparing initial conditions for MODIS LAI, AGB, ISCN SOC, and soil moisture across NA anchor sites.
 - Added GEDI AGB preparation workflow.
+- Added new feature of downloading NASA DAAC ORNL database.
```
Comment (mdietze) [inline CHANGELOG.md:51]: I think this log message needs to be more informative as you're not downloading the entire ORNL DAAC database

File: modules/data.remote/R/NASA_DAAC_download.R [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#'   downloaded files.Default to the working directory. If it doesn't exist it
+#'   will be created. Ignored if \code{just_path=TRUE}
+#' @param doi Character: data DOI on the NASA DAAC server.
+#' @param netrc_file Character: path to the credential file, default is NULL.
+#' @param just_path Boolean: if we just want the metadata and URL or proceed the actual download.
+#'
+#' @return A list containing meta data for each data downloaded.
+#' @export
+#' 
+#' @examples
```
Comment (mdietze) [inline modules/data.remote/R/NASA_DAAC_download.R:22]: would benefit from a "not run" example

File: modules/data.remote/R/NASA_DAAC_download.R [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+#' @param lr_lon Numeric: lower right longitude.
+#' @param ncore Numeric: numbers of core to be used if the maximum core
+#' @param from Character: date from which the data search starts. In the form
+#'   "yyyy-mm-dd".
+#' @param to Character: date on which the data search end. In the form
+#'   "yyyy-mm-dd".
+#' @param outdir Character: path of the directory in which to save the
+#'   downloaded files.Default to the working directory. If it doesn't exist it
+#'   will be created. Ignored if \code{just_path=TRUE}
+#' @param doi Character: data DOI on the NASA DAAC server.
```
Comment (mdietze) [inline modules/data.remote/R/NASA_DAAC_download.R:15]: This is probably the most important argument and would benefit from additional info so users know where to get these and how to format them. Would also be good to have a simple example below.

File: CHANGELOG.md (hunk 2 above)
Comment (DongchenZ) [reply CHANGELOG.md:51]: Fixed.

File: modules/data.remote/R/NASA_DAAC_download.R (hunk 4 above)
Comment (DongchenZ) [reply modules/data.remote/R/NASA_DAAC_download.R:15]: Fixed.

File: modules/data.remote/R/NASA_DAAC_download.R (hunk 3 above)
Comment (DongchenZ) [reply modules/data.remote/R/NASA_DAAC_download.R:22]: Added.

File: modules/data.remote/R/NASA_DAAC_download.R (hunk 1 above)
Comment (DongchenZ) [reply modules/data.remote/R/NASA_DAAC_download.R:29]: Fixed.

**Review by mdietze — APPROVED** [review]

**Pull request merged**